token.go

```diff
@@ -692,30 +695,67 @@ func processSingleResponse(ctx context.Context, sess *tdsSession, ch chan tokenS
 			ch <- order
 		case tokenDoneInProc:
 			done := parseDoneInProc(sess.buf)
+
+			ch <- done
 			if sess.logFlags&logRows != 0 && done.Status&doneCount != 0 {
-				sess.logger.Log(ctx, msdsn.LogRows, fmt.Sprintf("(%d row(s) affected)", done.RowCount))
+				sess.logger.Log(ctx, msdsn.LogRows, fmt.Sprintf("(%d rows affected)", done.RowCount))
+
+				if outs.msgq != nil {
+					_ = sqlexp.ReturnMessageEnqueue(ctx, outs.msgq, sqlexp.MsgRowsAffected{Count: int64(done.RowCount)})
+				}
+			}
+			if done.Status&doneMore == 0 {
+				if outs.msgq != nil {
+					// For now we ignore ctx->Done errors that ReturnMessageEnqueue might return
+					// It's not clear how to handle them correctly here, and data/sql seems
+					// to set Rows.Err correctly when ctx expires already
+					_ = sqlexp.ReturnMessageEnqueue(ctx, outs.msgq, sqlexp.MsgNextResultSet{})
+				}
+				return
 			}
-			ch <- done
 		case tokenDone, tokenDoneProc:
 			done := parseDone(sess.buf)
 			done.errors = errs
+			if outs.msgq != nil {
+				errs = make([]Error, 0, 5)
```

Comment:
Magic number 5 ?

Reply:
it's the same initial capacity set on line 668. I don't think the choice matters a whole lot.